.github/workflows/build.yml

```diff
@@ -0,0 +1,89 @@
+name: Build CI
+
+on:
+  push:
+    branches: [ master ]
```

Comment:
For a push vs a PR it would be nice if a different set of builds could be run to save time as pushes I assume can be less stringent / comprehensive.